### PR TITLE
add eksctl-anywhere

### PR DESCRIPTION
**Pull request by tuxtof**
<!--- Provide a general summary of your changes in the Title above -->

## Description
add eksctl-anywhere CLI 

solve #792 

## How Has This Been Tested?

```
make test                                                                                                                                                                                                                                       4s
CGO_ENABLED=0 go test github.com/alexellis/arkade github.com/alexellis/arkade/cmd github.com/alexellis/arkade/cmd/apps github.com/alexellis/arkade/cmd/chart github.com/alexellis/arkade/cmd/system github.com/alexellis/arkade/pkg github.com/alexellis/arkade/pkg/apps github.com/alexellis/arkade/pkg/archive github.com/alexellis/arkade/pkg/config github.com/alexellis/arkade/pkg/env github.com/alexellis/arkade/pkg/get github.com/alexellis/arkade/pkg/helm github.com/alexellis/arkade/pkg/k8s github.com/alexellis/arkade/pkg/types -cover
?   	github.com/alexellis/arkade	[no test files]
ok  	github.com/alexellis/arkade/cmd	(cached)	coverage: 17.2% of statements
ok  	github.com/alexellis/arkade/cmd/apps	(cached)	coverage: 1.9% of statements
?   	github.com/alexellis/arkade/cmd/chart	[no test files]
?   	github.com/alexellis/arkade/cmd/system	[no test files]
?   	github.com/alexellis/arkade/pkg	[no test files]
?   	github.com/alexellis/arkade/pkg/apps	[no test files]
?   	github.com/alexellis/arkade/pkg/archive	[no test files]
ok  	github.com/alexellis/arkade/pkg/config	(cached)	coverage: 22.9% of statements
?   	github.com/alexellis/arkade/pkg/env	[no test files]
ok  	github.com/alexellis/arkade/pkg/get	0.455s	coverage: 52.7% of statements
ok  	github.com/alexellis/arkade/pkg/helm	(cached)	coverage: 5.3% of statements
?   	github.com/alexellis/arkade/pkg/k8s	[no test files]
?   	github.com/alexellis/arkade/pkg/types	[no test files]
```

```
make e2e
...
=== RUN   Test_CheckTools/Download_of_eksctl-anywhere
    --- PASS: Test_CheckTools/Download_of_eksctl-anywhere (5.61s)
...
PASS
coverage: 54.9% of statements
ok  	github.com/alexellis/arkade/pkg/get	31.253s	coverage: 54.9% of statements
```

## Are you a GitHub Sponsor yet (Yes/No?)

<!-- Requests from sponsors take priority -->
<!--- Check at https://github.com/sponsors/alexellis         -->

- [ ] Yes
- [X] No

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`

<!--- In case it is a new application -->
- [] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for putting up a PR

> How Has This Been Tested?

Seems like you've not provided sufficient testing here so we can't merge until you do.

See a recent PR by @Jasstkn for what we're looking for.

**Comment by Jasstkn** [comment]
There is a [contributing guide](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) that I highly recommend to read. I think you will find all the answers there.

As a good example of expected changes for the PR like yours, check [this PR](https://github.com/alexellis/arkade/pull/769).

File: pkg/get/get_test.go [hunk 1]
```diff
@@ -1185,6 +1185,36 @@ func Test_DownloadEKSCTL(t *testing.T) {
...
 
+func Test_DownloadEKSCTLANYWHERE(t *testing.T) {
+	tools := MakeTools()
+	name := "eksctl-anywhere"
+
+	tool := getTool(name, tools)
+
+	const toolVersion = "v0.12.0"
+
+	tests := []test{
```
Comment (alexellis) [inline pkg/get/get_test.go:1196]: There are a bunch of missing unit tests here given how many downloads you have in the template?

File: pkg/get/tools.go [hunk 2]
```diff
@@ -647,6 +647,28 @@ https://github.com/inlets/inletsctl/releases/download/{{.Version}}/{{$fileName}}
...
+			BinaryTemplate: `
+			{{$os := .OS}}
+			{{$ext := "tar.gz"}}
+			{{$arch := .Arch}}
+			{{- if eq .Arch "x86_64" -}}
+			{{$arch = "amd64"}}
+			{{- else if eq .Arch "aarch64" -}}
+			{{$arch = "arm64"}}
+			{{- else if eq .Arch "arm64" -}}
+			{{$arch = "arm64"}}
```
Comment (alexellis) [inline pkg/get/tools.go:665]: I don't believe you are using this code path, try deleting it and see if your unit tests still work

File: pkg/get/get_test.go (hunk 1 above)
Comment (alexellis) [reply pkg/get/get_test.go:1196]: The template that you've added seems to have 3 different variations, so you may need to remove one,?

**Comment by alexellis** [comment]
Sorry, we need you to follow this step and show console output for both downloads:

```
Please always test your tools manually (hack/test-tool.sh) and with the end-to-end test tool (make e2e).
```

https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md

@Jasstkn could you make the instructions more explicit in the CONTRIBUTING.md file please? Perhaps that's the disconnect?

**Comment by tuxtof** [comment]
Make e2e is ok on my side but the hack test-tools not
Because you are force testing all the architecture even the one who are not released , that’s why is ask the question above

hack/test-tool.sh eksctl-anywhere
+ go build
+ ./arkade get eksctl-anywhere --arch arm64 --os darwin --quiet
Error: incorrect status for downloading tool: 404

**Comment by alexellis** [comment]
@tuxtof there is no forcing going on, sorry if you understood it that way.

Print the output of all the tools, the ones which are in the template you added.

**Comment by tuxtof** [comment]
@alexellis i clean the repo keeping only the working binary
also add the output of `make test` and `make e2e` in the description
`hack/test-tool.sh` is always not working as it try to DL arch who don't exist (see my comment above)

hope it will be ok for you

++

**Comment by Jasstkn** [comment]
@tuxtof hey. looks like your commits aren't signed. please, follow instructions from the bot's message.

**Review by Jasstkn — APPROVED** [review]
Hi! Thanks for the contribution. Looks good to me 🚀

**Comment by Jasstkn** [comment]
@alexellis hi. this PR looks good, you might want to take a look and accept it if anything is fine. 😎

**Comment by alexellis** [comment]
All good, thanks to you both :slightly_smiling_face:

**Pull request merged**